### ISSUE TITLE
the results are not expected when using the include='str', exclude='str' arguments in the transform

### ISSUE THREAD

**Issue by zzz123xyz** [fepegar/torchio#923]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug summary

For example, If I have several scans in the a subject such as subject['img'] subject['imgCT'], subject['label']. 
the include and exclude augments in transform will have unexpected behaviour. 

this is because in the function: 

class Subject(dict):
    def get_images_dict(
            self,
            intensity_only=True,
            include: Optional[Sequence[str]] = None,
            exclude: Optional[Sequence[str]] = None,
            ) -> Dict[str, Image]:
        images = {}
        for image_name, image in self.items():
            if not isinstance(image, Image):
                continue
            if intensity_only and not image[TYPE] == INTENSITY:
                continue
            if include is not None and image_name not in include:
                continue
            if exclude is not None and image_name in exclude:
                continue
            images[image_name] = image
        return images

in the line  "if include is not None and image_name not in include: " ,  if image_name = 'img' and include = 'imgCT'. 
the subject['img'] will still be included in the computation for the transform.

### Code for reproduction

```python
import torch
import torchio as tio

subject = tio.Subject(
  imgCT = tio.ScalarImage(os.path.join(case_path, 'imgCT.nii.gz')),
  img = tio.ScalarImage(os.path.join(case_path, 'img.nii.gz')),
  label = tio.LabelMap(os.path.join(case_path, 'segm.nii.gz'))
)
        
subject.plot()
transform = tio.Compose([
    tio.RandomAffine(degrees=(20, 30), exclude='img'),
    tio.RandomBlur(std=(3, 4), include='imgCT'),
])
transformed = transform(subject)
transformed.plot()
```

### Actual outcome

as described above
the subject['img'] will still be included in the computation for the transform.

### Expected outcome

the subject['img'] should not be included in the computation for the transform.

**Comment by ChristianHinge** [comment]
The include and exclude arguments should be lists of strings;

```
    tio.RandomAffine(degrees=(20, 30), exclude=['img']),
    tio.RandomBlur(std=(3, 4), include=['imgCT']),
```

Currently, the undesired behaviour arises from the following function in `src/torchio/transforms/data_parser.py`

    @staticmethod
    def _get_subject_from_dict(
        data: dict,
        image_keys: Sequence[str],
        label_keys: Optional[Sequence[str]] = None,
    ) -> Subject:
        subject_dict = {}
        label_keys = [] if label_keys is None else label_keys
        for key, value in data.items():
            if key in image_keys:
                class_ = LabelMap if key in label_keys else ScalarImage
                value = class_(tensor=value)
            subject_dict[key] = value
        return Subject(subject_dict)

On line 141 we have:

`if key in image_keys:`

which in @zzz123xyz's code would evaluate to:

```
if "img" in "imgCT"
> True
```
And when wrapping in list:

```
if "img" in ["imgCT"]
> False
```
@fepegar is this "pattern-matching" functionality an intended usecase when `include` / `exclude` are strings and not lists? If not, perhaps a quick type-check could be added to avoid confusion?

**Comment by fepegar** [comment]
Good catch, @ChristianHinge. I'm working on a fix in #1153.

**Comment by fepegar** [comment]
Thank you, both!
